amazon-kinesis-client/src/test/java/software/amazon/kinesis/lifecycle/ShardConsumerSubscriberTest.java

```diff
@@ -444,4 +448,167 @@ public void cancel() {
         }
     }
 
+    class TestShardConsumerSubscriber extends ShardConsumerSubscriber {
+
+        private int genericWarningLogged = 0;
+        private int readTimeoutWarningLogged = 0;
+        private RecordsPublisher recordsPublisher;
+        private ExecutorService executorService;
+        private int bufferSize;
+        private ShardConsumer shardConsumer;
+
+        TestShardConsumerSubscriber(int readTimeoutsToIgnoreBeforeWarning) {
+            this(mock(RecordsPublisher.class), Executors.newFixedThreadPool(1), 8, mock(ShardConsumer.class),
+                    readTimeoutsToIgnoreBeforeWarning);
+            Mockito.when(shardConsumer.shardInfo()).thenReturn(new ShardInfo("001", "token", null, null));
+        }
+
+        TestShardConsumerSubscriber(RecordsPublisher recordsPublisher, ExecutorService executorService, int bufferSize,
```

Comment:
The chained constructor is required for making initialization into the next constructor to pass into super. It's just to have a simple constructor that initializes the fields which gets passed into super and set during constructor.

It doesn't really matter if it happens in this simplified constructor or in the test method, but was from when I was using this class in different tests.

Reply:
Yeah lets remove this unnecessary constructor. We can have the same behavior in the other constructor. If needed in the future we can refactor and add the new constructor.

Reply:
No, it can't have this behavior in the other constructor. But I'll move the logic out of the constructors entirely. Since we cannot call anything before super, we cannot initialize class objects in a constructor without chaining.